### DISCUSSION TITLE
Is there a way to measure the instantaneous speed of an object?

### DISCUSSION THREAD

**Discussion by VictorBarbosa** [replit/kaboom#790] [General]
What I'm doing is, I have two platforms. On the first one, I have some "rockets" from where they will be launched, and on the other platform, they must "land." However, I need to know the speed at which the rocket reaches the platform because if it's too fast, it should explode.

The larger platform is where it takes off, and the smaller platform is where it should land. I will implement a neural network to train and solve this; that's why I can't simply create an algorithm that starts a timer when the onKeyDown event is triggered and stops it when the onKeyRelease event is triggered. However, I need to know the speed.

![Captura de Tela 2023-11-22 às 8 04 32 AM](https://github.com/replit/kaboom/assets/6170478/e609e2a4-7bd6-40d7-bba6-8dc786664e55)

**Comment by slmjkdbtl** [comment]
If you're using `body()` component you can get `rocket.vel` which gives you a `Vec2` of its current velocity. It's available since v3000.1.17.

Even without this, you can use `onUpdate()` and keep track of object's position each frame and calculate the difference to get the velocity last frame

**Reply by VictorBarbosa** [reply]
Thank you
